Taskfile.yml

```diff
@@ -326,8 +327,16 @@ tasks:
     deps: [controller:generate-kustomize]
     dir: "v2/"
     cmds:
+    - mkdir -p config/crd/packed
+    - kustomize build config/default > config/crd/packed/azureserviceoperator.yaml
```

Comment:
This directory seems like a strange place to put this file - it's not just CRDs, if you're building `config/default` the output will include the deployment, webhook service, RBAC, webhook configurations, etc.

It might make more sense just to run `kustomize build config/crd`, then the output would only be the patched CRDs. (You wouldn't be able to use this file as a precursor to building the full release yaml, but I think that's ok.)

Also, I'm fairly sure building `config/default` will mean that the conversion sections are all pointing at the `azureserviceoperator-system` namespace, which doesn't seem right - the mutating/validating webhook configurations that get included from `../config/webhook` won't have those prefixes.

So maybe leave the `run-kustomize` task the way it was, and change `pack-crds` to just build `config/crd`?

Reply:
I need the output as a file so that it can be picked up by our tests, which expect to point at an existing file.

Should I move this file to a different location or change the target I'm building?

Reply:
@babbageclunk just trying to make sure I understand what you are saying... you are saying:

1. Change `run-kustomize` back to the way it was before. This target will run kustomize at the `config/default` level and produce a file containing _everything_ (webhooks, crds, deployment, etc). Primary usage for this target is when producing the official release and/or creating a deployment to deploy into envtest (or a real cluster).
2. Create a new target (@theunrepentantgeek you have `pack-crd` now but maybe it could be called `produce-envtest-crds` or `run-kustomize-envtest` or something). This target would _just_ produce a file containing CRDs + conversion webhooks for use in envtest. Doing this involves running kustomize on just the `config/crd` directory rather than `config/default`?

I did look into the envtest code and confirm that if envtest is passed a file with stuff in it other than CRDs for the `CRDDirectoryPaths` parameter it just ignores everything that isn't a CRD (at least, that was my read of their code).

Reply:
If that is what you're saying, I think I like that idea? If we don't do that I think we should rename this `pack-crd` command as all it's really doing is replacing an already generated YAML's docker image reference, so it feels like `pack-crd` should be called something like `prepare-for-release` or somesuch.

Reply:
@matthchr Yup, that's what I was saying - sorry it wasn't too clear! 

> I did look into the envtest code and confirm that if envtest is passed a file with stuff in it other than CRDs for the CRDDirectoryPaths parameter it just ignores everything that isn't a CRD (at least, that was my read of their code).

I'm sure this is right, but I think the namespace mangling that is done by the `config/default` kustomize file would still mean the conversion webhooks pointed at the wrong place.

Reply:
Moved output into `crd/out` for clarity and made the suggested change.